services/venice-controller/src/main/java/com/linkedin/venice/controller/server/StoresRoutes.java

```diff
@@ -431,6 +432,15 @@ public void internalHandle(Request request, StoreMigrationResponse veniceRespons
           veniceResponse.setErrorType(ErrorType.BAD_REQUEST);
           return;
         }
+        VeniceControllerClusterConfig destClusterConfig = admin.getControllerConfig(destClusterName);
+        // Both source and destination clusters should either have RT versioning enabled or disabled
+        if (destClusterConfig != null && (admin.getControllerConfig(srcClusterName)
+            .isRealTimeTopicVersioningEnabled() != destClusterConfig.isRealTimeTopicVersioningEnabled())) {
+          veniceResponse
+              .setError("Source cluster and destination cluster both should have RT versioning enabled or disabled ");
+          veniceResponse.setErrorType(ErrorType.BAD_REQUEST);
+          return;
+        }
```

Comment:
Could you please add a log here when destination cluster config is not found?

Reply:
added